### PR TITLE
MMCA-4278 Update XI EORI address information as per the latest spec c…

### PR DESCRIPTION
**Pull request by harshare**
…hanges

### CODE REVIEW

**Review by AdamCoops — APPROVED** [review]

File: app/models/XiEoriAddressInformation.scala [hunk 1]
```diff
@@ -18,11 +18,11 @@ package models
 
 import play.api.libs.json.{Json, OFormat}
 
-case class XiEoriAddressInformation(streetNumber1: String,
-                                    streetNumber2: Option[String],
-                                    city: String,
-                                    countryCode: String,
-                                    postalCode: Option[String])
+case class XiEoriAddressInformation(pbeAddressLine1: String,
+                                    pbeAddressLine2: Option[String],
```
Comment (tyagikumarvineet) [inline app/models/XiEoriAddressInformation.scala]: Could you please update pbeAddressLine2, pbeAddressLine3, pbeAddressLine4 and pbePostCode to have the default value as None and after that we we do not need to specify the value like Some("") or None (if you do not want to use those empty values specifically) at line 40 of XiEoriController.scala, that can be updated to as below

xiEoriInformationRepository.set(eori, XiEoriInformation("", "", XiEoriAddressInformation(""))) or if we want to make it more readable then better to say 

          xiEoriInformationRepository.set(
            eori,
            XiEoriInformation("","", XiEoriAddressInformation(pbeAddressLine1 = ""))

File: app/controllers/XiEoriController.scala [hunk 2]
```diff
@@ -37,7 +37,7 @@ class XiEoriController @Inject() (xiEoriInformationRepository: XiEoriInformation
       case Some(xiEoriInformation) => Future.successful(Ok(Json.toJson(xiEoriInformation)))
       case None => retrieveXiEoriInformation(eori).map {
         case Some(xiEoriInformation) => Ok(Json.toJson(xiEoriInformation))
-        case None => xiEoriInformationRepository.set(eori, XiEoriInformation("", "", XiEoriAddressInformation("",None,"","", None)))
+        case None => xiEoriInformationRepository.set(eori, XiEoriInformation("", "", XiEoriAddressInformation("",Some(""),Some(""),None, None)))
```
Comment (tyagikumarvineet) [inline app/controllers/XiEoriController.scala]: Could you please split the line (as below) over multiple lines as it has become very long (let me know if its not showing long on your machine)? Also better to add a val like below at the start of the method and use in place of ""

val emptyString = ""

          xiEoriInformationRepository.set(
          eori,
          XiEoriInformation(emptyString, emptyString, XiEoriAddressInformation(pbeAddressLine1 = emptyString)))

File: app/models/XiEoriAddressInformation.scala (hunk 1 above)
Comment (harshare) [reply app/models/XiEoriAddressInformation.scala]: Updated default value as None to optional address fields.

File: app/controllers/XiEoriController.scala (hunk 2 above)
Comment (harshare) [reply app/controllers/XiEoriController.scala]: Made changes to split into multiple lines and added an emptystring val.

**Review by tyagikumarvineet — APPROVED** [review]

**Pull request merged**